assets/js/blocks/reviews-by-category/edit.js

```diff
@@ -0,0 +1,315 @@
+						max={ maxPerPage }
+						min={ minPerPage }
+					/>
+					<ToggleControl
+						label={ __( 'Load more', 'woo-gutenberg-products-block' ) }
+						checked={ attributes.showLoadMore }
+						onChange={ () => setAttributes( { showLoadMore: ! attributes.showLoadMore } ) }
+					/>
+					{ attributes.showLoadMore && (
+						<RangeControl
+							label={ __( 'Load More Reviews', 'woo-gutenberg-products-block' ) }
+							value={ attributes.reviewsOnLoadMore }
+							onChange={ ( reviewsOnLoadMore ) => setAttributes( { reviewsOnLoadMore } ) }
+							max={ maxPerPage }
+							min={ minPerPage }
+						/>
+					) }
+				</PanelBody>
+			</InspectorControls>
+		);
+	};
+
+	const renderEditMode = () => {
+		const onDone = () => {
+			setAttributes( { editMode: false } );
+			debouncedSpeak(
+				__(
+					'Showing Reviews by Category block preview.',
+					'woo-gutenberg-products-block'
+				)
+			);
+		};
+
+		return (
+			<Placeholder
+				icon={ <IconReviewsByCategory className="block-editor-block-icon" /> }
+				label={ __( 'Reviews by Category', 'woo-gutenberg-products-block' ) }
+				className="wc-block-reviews-by-category"
+			>
+				{ __(
+					'Show product reviews from specific categories.',
+					'woo-gutenberg-products-block'
+				) }
+				<div className="wc-block-reviews-by-category__selection">
+					<ProductCategoryControl
+						selected={ attributes.categoryIds }
+						onChange={ ( value = [] ) => {
+							const ids = value.map( ( { id } ) => id );
+							setAttributes( { categoryIds: ids } );
+						} }
+					/>
+					<Button isDefault onClick={ onDone }>
+						{ __( 'Done', 'woo-gutenberg-products-block' ) }
+					</Button>
+				</div>
+			</Placeholder>
+		);
+	};
+
+	const renderViewMode = () => {
+		const showReviewImage = ( showAvatars || attributes.imageType === 'product' ) && attributes.showReviewImage;
+		const showReviewRating = enableReviewRating && attributes.showReviewRating;
+
+		if ( ! showReviewContent && ! showReviewRating && ! showReviewDate && ! showReviewerName && ! showReviewImage && ! showProductName ) {
+			return (
+				<Placeholder
+					className="wc-block-reviews-by-category"
+					icon={ <IconReviewsByCategory className="block-editor-block-icon" /> }
+					label={ __( 'Reviews by Category', 'woo-gutenberg-products-block' ) }
+				>
+					{ __( 'The content for this block is hidden due to block settings.', 'woo-gutenberg-products-block' ) }
+				</Placeholder>
+			);
+		}
+
+		const classes = classNames( 'wc-block-reviews-by-category', className, {
+			'has-image': showReviewImage,
+			'has-name': showReviewerName,
+			'has-date': showReviewDate,
+			'has-rating': showReviewRating,
+			'has-content': showReviewContent,
+			'has-product-name': showProductName,
+		} );
+
+		return (
+			<div className={ classes }>
+				<EditorBlock attributes={ attributes } />
+			</div>
+		);
+	};
+
+	if ( ! categoryIds || editMode ) {
+		return renderEditMode();
+	}
+
+	return (
```

Comment:
In _Reviews by Product_ we have error and loading states handling. Should we copy it here?

Reply:
No, there is no API call in this block. Since it supports multiple categories I chose not to get details about all selected categories to avoid lots of extra requests.